service/create/k8s_secret.go

```diff
@@ -0,0 +1,54 @@
+package create
+
+import (
+	"github.com/giantswarm/certificatetpr"
+	microerror "github.com/giantswarm/microkit/error"
+	"k8s.io/client-go/pkg/api/errors"
+	"k8s.io/client-go/pkg/api/v1"
+)
+
+// SaveCertificate saves the certificate as a k8s secret.
+func (s *Service) SaveCertificate(cert certificateSecret) error {
+	var err error
+
+	secret := &v1.Secret{
+		ObjectMeta: v1.ObjectMeta{
+			Name: cert.CommonName,
+		},
+		StringData: map[string]string{
+			"crt": cert.IssueResponse.Certificate,
+			"key": cert.IssueResponse.PrivateKey,
+			"ca":  cert.IssueResponse.IssuingCA,
+		},
+	}
+
+	// Create the secret.
+	_, err = s.Config.K8sClient.Core().Secrets(cert.Namespace).Create(secret)
+	if errors.IsAlreadyExists(err) {
+		// Update the secret if it already exists.
+		_, err = s.Config.K8sClient.Core().Secrets(cert.Namespace).Update(secret)
```

Comment:
If the secret already exists then I'm saving the updated cert. But I'm not sure this is right? It could also be an error state as it shouldn't happen if the watch is working correctly.